src/HttpServer/HttpServer.cc

```diff
@@ -80,10 +82,14 @@ void HttpServer::RegisterRoutes() {
 }
 
 void HttpServer::HandleGetConfig(Res* res, Req* req) {
-    json runtime_config = {{"apiAddress", "/api"}};
-    res->writeStatus(HTTP_200);
-    res->writeHeader("Content-Type", "application/json");
-    res->end(runtime_config.dump());
+    if (_enable_runtime_config) {
+        json runtime_config = {{"apiAddress", "/api"}};
+        res->writeStatus(HTTP_200);
+        res->writeHeader("Content-Type", "application/json");
+        res->end(runtime_config.dump());
+    } else {
+        res->writeStatus(HTTP_200)->end();
+    }
```

Comment:
I think this should be enabled or disabled in `RegisterRoutes`, not here, and return 501 (not implemented) for consistency with other disabled features (would that require a change in the calling code?).

If there's a good reason for it to return 200 anyway, I think we should still handle this logic in `RegisterRoutes` and make a different default handler (called something like `QuietSuccess`, or something).

Reply:
There's a good reason to return 200, because it just means there's no backend-mandated config. I can add it to a separate handler